write.go

```diff
@@ -96,8 +96,16 @@ func writeFileHeader(w dicomio.Writer, ds *Dataset, metaElems []*Element, opts .
 	tagsUsed[tag.FileMetaInformationGroupLength] = true
 
 	// TODO make better structure for error checking so it's no so many lines
-	// TODO Optionally write FileMetaInformationVersion before these required headers to maintain order
-	err := writeMetaElem(subWriter, tag.MediaStorageSOPClassUID, ds, &tagsUsed, opts...)
+	// XXXHACK TODO: decide how to handle this header, since it needs to be written here for correct ordering
+	_, err := ds.FindElementByTag(tag.FileMetaInformationVersion)
+	if err != ErrorElementNotFound {
```

Comment:
is this intended? This will only execute this code if the error is something other that Element not found, which could include other errors in the future. I expect if there is any error here in finding the tag, we do not want to write this (and probably want to handle it in a specific way if needed).

Also I don't think this is needed, because writeMetaElem does this existence checking within it already?

Also, this tag is found in metaElems already right? I'm unsure if this function needs to take both metaElems and ds *Dataset, but I can clean this and other things up in a future PR as well.

Reply:
You're right, my logic in this block of code wasn't entirely sound. I've changed this code block to just try and write the header (as you mentioned the writeMetaElem does the existence checking) and then I handle the error afterwards. If the error is anything but ErrorElementNotFound then I return the error, otherwise I ignore it move on to the next one